cpp/csp/adapters/websocket/WebsocketEndpoint.cpp

```diff
@@ -49,23 +58,57 @@ void WebsocketEndpoint::run()
         );
     }
     m_session->run();
+}
+
+WebsocketEndpoint::~WebsocketEndpoint() {
+    try {
+        // Call stop but explicitly pass false to prevent io_context shutdown
+        stop(false);
+    } catch (...) {
+        // Ignore any exceptions during cleanup
+    }
+}
 
-    m_ioc.run();
+void WebsocketEndpoint::stop( bool stop_ioc )
+{
+    if( m_session ) m_session->stop(); 
+    if( stop_ioc ) m_ioc.stop();
 }
 
-void WebsocketEndpoint::stop()
-{ 
-    m_ioc.stop();
-    if(m_session) m_session->stop(); 
+void WebsocketEndpoint::updateHeaders(csp::Dictionary properties){
+    DictionaryPtr headers = m_properties->get<DictionaryPtr>("headers");
+    for (auto it = properties.begin(); it != properties.end(); ++it) {
+        std::string key = it.key();
+        auto value = it.value<std::string>();
+        headers->update(key, std::move(value));
+    }
 }
 
+void WebsocketEndpoint::updateHeaders(const std::string& properties) {
+    if( properties.empty() )
+        return;
+    DictionaryPtr headers = m_properties->get<DictionaryPtr>("headers");
+    rapidjson::Document doc;
+    doc.Parse(properties.c_str());
```

Comment:
what is the reasoning behind using a serialized version of the headers here versus a dictionary (or `DictionaryPtr`)?

Reply:
No ts support in cpp, whereas string does have ts support. May change down the road if we implement a headers struct